### PR TITLE
Apply Persistent Subscriptions RFC - 022.

### PR DESCRIPTION
**Pull request by YoEight**

### CODE REVIEW

**Comment by YoEight** [comment]
@ylorph samples added!

**Comment by ylorph** [comment]
I'm missing or missed in the sample
CreatePersistentSubscription
ConnectToPersistentSubscriptionToAll ( i.e no manual ACK)

( those are manily to have the correcect samples to point to from the docs )

File: esdb/persistent_options.go [hunk 1]
```diff
@@ -2,28 +2,28 @@ package esdb
...
+	if o.StartFrom == nil {
+		o.StartFrom = End{}
 	}
 }
 
 type PersistentAllSubscriptionOptions struct {
 	Settings           *SubscriptionSettings
-	From               AllPosition
+	StartFrom          AllPosition
 	MaxSearchWindow    int
```
[the remark this answers is not in the document]
Comment (YoEight) [reply esdb/persistent_options.go:18]: It's used to configure the filter.

**Comment by YoEight** [comment]
@ylorph I added the missing create persistent subscription sample.

File: esdb/persistent_options.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ylorph) [reply esdb/persistent_options.go:18]: I'm not seing  CheckpointInterval  as a specific setting for filter in the  .net client

**Review by ylorph — APPROVED** [review]

File: esdb/types.go [hunk 2]
```diff
@@ -581,3 +581,43 @@ func ExcludeSystemEventsFilter() *SubscriptionFilter {
...
+	PersistentSubscriptionStatus_ReplayingParkedMessages = "ReplayingParkedMessages"
+	PersistentSubscriptionStatus_Live                    = "Live"
+)
+
+type PersistentSubscriptionInfo struct {
+	EventStreamId            string                        `json:"eventStreamId"`
+	GroupName                string                        `json:"groupName"`
+	Status                   PersistentSubscriptionStatus  `json:"status"`
+	AverageItemsPerSecond    float64                       `json:"averageItemsPerSecond"`
+	TotalItemsProcessed      int64                         `json:"totalItemsPerSecond"`
```
Comment (pvanbuijtene) [inline esdb/types.go:600]: Is `totalItemsPerSecond` different for a reason?

**Review by pvanbuijtene — CHANGES_REQUESTED** [review]

**Review by pvanbuijtene — APPROVED** [review]

**Review by thefringeninja — APPROVED** [review]

**Pull request merged**